### PR TITLE
TCP transport

### PR DESCRIPTION
**Pull request by rneswold**
This branch contains the proposed changes to support TCP connections to DPM. This is slated to be part of v1.0.0 of the package, so it needs to work as well as the ACNET version did in v.0x.

To-do list:

- [x] Run more aggressive scripts against it
- [x] Propose API changes because this is targeted for v1.x and we don't want to make breaking changes anytime soon. No, we're going to run with this API.
- [x] Use `protobufs` for the protocol (see previous concerns)? No, we'll switch over at a later time. The users are mostly protected from the raw protocol messages so updating later on, theoretically, shouldn't show changes to them.
- [x] Point the package to `acsys-proxy.fnal.gov` (092d941).

### CODE REVIEW

File: acsys/dpm/__init__.py [hunk 1]
```diff
@@ -6,109 +6,90 @@
...
+                                    ApplySettings_reply,
                                     Authenticate_request,
                                     EnableSettings_request,
-                                    TimedScalarArray_reply, Authenticate_reply)
+                                    TimedScalarArray_reply,
+                                    Authenticate_reply,
+                                    unmarshal_reply)
 
-_log = logging.getLogger(__name__)
+_log = logging.getLogger('acsys')
```
Comment (beauremus) [inline acsys/dpm/__init__.py]: This, https://docs.python.org/3.6/library/logging.html#logger-objects, recommends using `__name__` for modules to match the package hierarchy. Do you have an idea of why that doesn't work for us?

File: acsys/dpm/__init__.py [hunk 2]
```diff
@@ -144,113 +131,206 @@ class ItemStatus(_ItemCommon):
...
-        # fatal errors percolate up.
-
-        if exception != acsys.status.ACNET_UTIME:
-            raise
-    return result
-
-
-class DPM:
-    def __init__(self, con, node):
+    def __init__(self):
```
Comment (beauremus) [inline acsys/dpm/__init__.py]: Does `DPM` need `super().__init__()`? I see that it's not in the documentation examples, but from everything else, I know it should be used to inherit.

File: acsys/dpm/__init__.py [hunk 3]
```diff
@@ -287,157 +366,97 @@ def _xlat_reply(self, msg):
...
-            # status. Whatever it was, we need to pick another DPM and
-            # add all the current requests.
-
-            _log.warning('DPM(id: %s) connection closed ... retrying',
-                         str(self.list_id))
-            await self._restore_state()
+    async def connect(self):
+        loop = asyncio.get_event_loop()
+        con_fut = loop.create_connection(lambda: self,
+                                         host='dce46.fnal.gov',
```
Comment (beauremus) [inline acsys/dpm/__init__.py]: Just a reminder that we need to change this to `acsys-proxy` before merging.

**Review by beauremus — CHANGES_REQUESTED** [review]
This looks good. As you mentioned we should test it some more. I'm flagging this review as "request changes" so that we can modify the end point before merging.

File: acsys/dpm/__init__.py (hunk 1 above)
Comment (rneswold) [reply acsys/dpm/__init__.py]: I was being a bit lazy. In my test scripts, I get the `acsys` logger and set the level. I guess the test scripts can grab each `logger` for each sub package and set the level.

File: acsys/dpm/__init__.py (hunk 2 above)
Comment (rneswold) [reply acsys/dpm/__init__.py]: I don't specify a base class so I didn't think it was necessary.

File: acsys/dpm/__init__.py (hunk 3 above)
Comment (rneswold) [reply acsys/dpm/__init__.py]: I added it to the checklist in the pull request.

File: acsys/dpm/__init__.py (hunk 1 above)
Comment (rneswold) [reply acsys/dpm/__init__.py]: Fixed in e3725eb.

File: acsys/dpm/__init__.py (hunk 2 above)
Comment (beauremus) [reply acsys/dpm/__init__.py]: Is `asyncio.Protocol` not the base class?

Comment (rneswold) [reply acsys/dpm/__init__.py]: You're right; I was just looking at the snippet above. To be safe, I'll add the `super.init()` call.

Comment (rneswold) [reply acsys/dpm/__init__.py]: Fixed: d6de4b7

**Review by beauremus — APPROVED** [review]

**Pull request merged**